src/client/Item/Item.ts

```diff
@@ -28,7 +29,7 @@ export class Item {
   constructor(
     public readonly container: Container,
     public readonly id: string,
-    public readonly partitionKey: string,
+    public partitionKey: PartitionKey,
```

Comment:
We shouldn't make this public settable. We should use a _partitionKey or partitionKeyInternal with a getter instead.

Reply:
I made it private and it only gets public set in the constructor